### PR TITLE
Fix #339 - Fixing the issue of not handling the procedures's options

### LINKED ISSUE #339: Bad procedure body
```sql
 CREATE DEFINER=`root`@`%` PROCEDURE `test2`(IN `_var` INT) NOT DETERMINISTIC NO SQL SQL SECURITY INVOKER NO SQL SQL SECURITY INVOKER SELECT _var
 ```
 
The detected is 
```sql
NO SQL
    SQL SECURITY INVOKER
SELECT _var
```
It should be
```sql
SELECT _var
```

Ref: https://github.com/phpmyadmin/phpmyadmin/issues/15991

### PR DESCRIPTION
**Pull request by iifawzi**
Hi, This PR should fix the issues relating to #339
The issue was in https://github.com/phpmyadmin/sql-parser/blob/2d1efdec77339bfbe0ee1ac573ea672887c34421/src/Statements/CreateStatement.php#L221-L237
it had some wrong tokens, and wasn't have many, which was leading to break the for loop that parses the options tokens
https://github.com/phpmyadmin/sql-parser/blob/2d1efdec77339bfbe0ee1ac573ea672887c34421/src/Components/OptionsArray.php#L126-L129

I've fixed and added the correct options tokens as much as I could. [Ref](https://dev.mysql.com/doc/refman/8.0/en/create-procedure.html)

Signed-off-by: Fawzi E. Abdulfattah <iifawzie@gmail.com>

### CODE REVIEW

**Comment by iifawzi** [comment]
I've made a fake integration with PMA, and all the tests passed too :')

File: tests/Builder/CreateStatementTest.php [hunk 1]
```diff
@@ -482,9 +482,7 @@ public function testBuilderCreateFunction()
         $this->assertSame(
             'CREATE DEFINER=`root`@`localhost`'
             . ' FUNCTION `inventory_in_stock` (`p_inventory_id` INT) RETURNS TINYINT(1)'
-            . ' READS SQL DATA' . "\n"
```
Comment (williamdes) [inline tests/Builder/CreateStatementTest.php:485]: This is not correct, the builder needs a patch

File: tests/Builder/CreateStatementTest.php [hunk 2]
```diff
@@ -407,14 +407,14 @@ public function testBuilderCreateProcedure()
...
+            . ' PROCEDURE `test2` (IN `_var` INT) '
+            . ' SELECT _var',
             $stmt->build()
         );
 
-        $this->assertTrue($stmt->entityOptions->isEmpty());
+        $this->assertFalse($stmt->entityOptions->isEmpty());
         $this->assertFalse($stmt->options->isEmpty());
-
+
```
Comment (williamdes) [inline tests/Builder/CreateStatementTest.php:417]:
```suggestion

```

**Review by williamdes — APPROVED** [review]
Perfect

File: tests/Parser/AlterStatementTest.php [hunk 3]
```diff
@@ -52,6 +52,7 @@ public function alterProvider()
             array('parser/parseAlterUser5'),
             array('parser/parseAlterUser6'),
             array('parser/parseAlterUser7'),
+            array('parser/parseAlterUser8'),
```
Comment (williamdes) [inline tests/Parser/AlterStatementTest.php:55]: Good catch, I forgot to let you know I had patched it in: 1acb64c8b2bc98b4c9741d34475a7d06c40eb861

**Pull request merged**